src/ast/passes/semantic_analyser.cpp

```diff
@@ -296,6 +296,8 @@ class SemanticAnalyser : public Visitor<SemanticAnalyser> {
   // Holds the function argument index currently being
   // visited by this SemanticAnalyser.
   int func_arg_idx_ = -1;
+  // typeof, sizeof, cast, offsetof
+  bool is_type_func_ = false;
```

Comment:
IIUC, setting this means that the currently visited node should be interpreted as a data type. It took me a while to understand that so I think we could come up with a better name for this field. How about `is_type_name_`?